packages/features/form-builder/Components.tsx

```diff
@@ -83,9 +88,75 @@ export const Components: Record<BookingFieldType, Component> = {
     factory: (props) => <Widgets.NumberWidget noLabel={true} {...props} />,
   },
   name: {
-    propsType: "text",
+    propsType: "variants",
     // Keep special "name" type field and later build split(FirstName and LastName) variant of it.
-    factory: (props) => <Widgets.TextWidget noLabel={true} {...props} />,
+    factory: (
+      props: Omit<TextLikeComponentProps, "value" | "setValue"> & {
+        variant: "firstAndLastName" | "fullName";
+        variants: NonNullable<z.infer<typeof fieldsSchema>[number]["variantsConfig"]>["variants"];
+        value: Record<string, string>;
+        setValue: (value: string | Record<string, string>) => void;
+      }
+    ) => {
+      const { variant: variantName = "fullName" } = props;
+      const onChange = (name: string, value: string) => {
+        props.setValue({
+          ...props.value,
+          [name]: value,
+        });
+      };
+      if (!props.variants) {
+        throw new Error("'variants' is required for 'name' type of field");
+      }
+
+      if (variantName === "fullName") {
+        const variant = props.variants[variantName];
+        const variantField = variant.fields[0];
+        let value;
+        if (typeof props.value !== "string") {
+          value = "";
+        } else {
+          value = props.value;
+        }
+        return (
+          <InputField
```

Comment:
We can maybe make a component BookingInputField local to this file that would have the same props required by all input fields.